tt_metal/Glossary.md

```diff
@@ -0,0 +1,145 @@
+- **HBM -** High Bandwidth Memory
+- **HPC -** High Performance Computing
+- **I2S - S2I -** Interleaved to Sharded, Sharded to Interleaved.
+- **ISA -** Instruction-Set Architecture - An abstract computer model that determines how the CPU is controlled by the software.
+- **ISS -** Instruction Set Simulator
+- **JVM -** Java Virtual Machine
+- **KWARG -** Keyword Argument - used client side to allow all arguments, not used in server side functions.
+- **LLK -** Low Level Kernels - Bottom layer of software stack, responsible for basic system operations.
+- **LLM -** Large Language Model
+- **LPDDR4 -** DRAM - Off-chip Memory. Provides larger, off-chip storage for the system.
+- **LUT -** Look Up Table - Mapping table of input and output values used for addressing.
+- **MLP -** Multilayer Perceptron - A modern feedforward artificial neural network.
+- **MLIR -** Multi-Level Intermediate Representation
+- **MM - MatMul - Matrix Multiplication -** Multiplication of two square or rectangular matrices.
+- **MIMD -** Multiple Instruction Multiple Data - Allows multiple processors to function simultaneously and asynchronously.
+- **NM -** Nanometers
+- **NMC -** Near Memory Computing - Computing paradigm that moves data processing closer to where the data is stored.
+- **NLP -** Natural Language Processing
+- **NoC -** Network on Chip
+- **NPU -** Neural Processing Unit
+- **NUMA -** Non-uniform Memory Access
+- **PCC -** Pearson Correlation Coefficient - A method of measuring a linear correlation between two variables.
+- **Q, K, V values -** Q, K, and V stand for different values. Q = query, K = key, V = value. In transformer architecture these values are used to help the model focus on relevant parts of the input sequence.
+- **RB -** Routing Buffer
+- **RDMA -** Remote Direct Memory Access
+- **SDK -** Software Development Kit
+- **seqL -** Sequence Length
+- **SFPU -** Sundance Flash Programming Utility
+- **SIMD -** Single Instruction Multiple Data - Allows processors to execute the same instruction on multiple data points simultaneously
+- **SoC -** System on Chip
+- **SRAM -** Static Random Access Memory. On-chip Memory. 1 MB SRAM memory (L1), a scratch pad accessible by all RISC-V processors and engines within the core.
+- **TM -** Tensor Movement
+- **TP -** Tensor Parallel
+- **ViT -** Vision Transformer - A transformer model used for vision processing tasks.
+- **VM -** Virtual Machine
+## **Terminology** ##
+- **ARC Core -** Manages board-level control and administration.
+- **Argument -** A value or set of values as an input that allows customization of code functions.
+- **Atomic Operation -** Sequences of operations completed without interruption. Atomic operations include counter updates and other data integrity tasks.
+- **Baby RISC-V -** Miniature RISC-V chips used in Tensix Cores.
+- **Bool -** Boolean data type.
+- **Convolution -** A mathematical operation on two functions that produces a third function.
```

Comment:
Not sure what this means. I think you should describe convolution in the context of image processing, not mathematics.